### DISCUSSION TITLE
【ブレスト】LGTM画像のDLページのデザイン案を考える

### DISCUSSION THREAD

**Discussion by kagomen** [first-contributions-ja/first-contributions-ja.github.io#178] [00: 未分類]
## 現時点で考えられる案
- SNSでシェアしやすいようにシェアボタンを設置
- ファスコン参加者への明るいメッセージを記載
- ファスコンのデザインを受け継ぐ感じ？

何か入れたい情報や要素があればお気軽に提案お願いします！！

## 関連リンク
#173

**Comment by kazzyfrog** [comment]
ダウンロードできる個別ページがあるなら、やっぱり複数あるテーマを選ぶとか、編集できるとかあってもいいかもな、と思いました！

ファスコンのチュートリアルに参加した人の、さらに実践的なコントリビューションを提示できるので、結局これも「さらなるcontribution」としてハマるかも！
（とはいえ、HTML, CSSが書ければ、LGTMのテーマ作成できるので簡単！）

**Reply by kazzyfrog** [reply]
チュートリアルで入力する、データを画面上でどんな感じで確認することもできるので、
絵文字のフォントが微妙に違うとか、好きな色が黒とか白でも、ちゃんと文字が表示できるかが、少し懸念点でもあるので！

あと、個人的には、自分が継続的に使いたいサービスになりそうだなと思いました！
色々テーマあったり、チュートリアルで入力した値を変更することができたら、飽きずに使えるかなと！
気が向いた時に、自分たちでも追加する感じ

**Reply by kagomen** [reply]
個別ページで直接編集ができる仕様は、

- やっぱり「別のアプリ（LGTM画像生成アプリ）感」がある
- 入力してもらった直後に簡単に編集できてしまうと、入力したプロフィールデータがあんまり意味を成さない気がする

という点で、どうなんだろう…？とは思うのですが、

> ファスコンのチュートリアルに参加した人の、さらに実践的なコントリビューションを提示できる

> 「さらなるcontribution」としてハマるかも！

> 色々テーマあったり

こちらはすごく良いなー！と思いました！
なので（？）、kazzyfrogさんのアイデアを踏まえつつ自分が考えたのは、以下のような感じです。

DLページにて、

```
First Contributions JAへのご参加、ありがとうございました！
〜うんたらかんたら〜
生成されたLGTMステッカーは気に入りましたか？
もし、「ステッカーをもっとカスタマイズしたい！」とお考えであれば、コントリビューションのさらなるステップとして、自作テーマを作成することも可能です！
また、作成した自作テーマは、他の方にも使っていただけます！
```

のような文言を記載して、自作テーマを追加する方法を書いたmdファイルに誘導する。

というのは、どうでしょうか……？
ファスコンはやっぱり「共同開発体験」ができてこそ！だと思うので、LGTM画像を手軽に編集してもらうサービス（＝LGTM画像生成サービス）よりも、こういう形で体験を提供（？）できた方が、良いのではないかなーと

ご意見うかがいたいです！

**Reply by kagomen** [reply]
上記のDLページのデザインのイメージは、kazzyさんのデザインカンプの、左側と中央だけある、という感じです
自作テーマが追加されるごとに、左側の自作テーマの欄も増えていって、クリックしたら真ん中のLGTM画像がそのテーマのLGTM画像に切り替わるって感じです（伝わりますか…？）

色や絵文字はせっかく選んでもらったから、変更できない方が良いんじゃないかな？という気が、個人的にはします！（変更したい場合は、Contributions.jsonを編集→変更してもらう形でもいいのかな、と！）

**Reply by kagomen** [reply]
![image](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/154225199/e69f0740-a5e3-464a-943d-295b376b82f9)

**Reply by kagomen** [reply]
## 追記
- 自作テーマを開発したらみんながそのテーマを自由に使えるのは、OSSっぽさもあって、すごく良いなと思いました！
- DLページで画像編集できないようにした方が、自作テーマ開発に流れてくれそうかなと思いました！

**Reply by kazzyfrog** [reply]
確かにそうですね！

DLページで画像編集できる！のは、一旦無視していいかもです！

個人的に以下のようなことが起きそうかも、と思ったのですが、
まあそういうことが起きたら考えればよさそうですね！

- テーマによって、好きな色が白とか黒だと、背景や文字の色に使われて、周りの色と同化して見えなくなるとか
- Contributions.jsonには、好きな絵文字を１つ入力して、複数の絵文字を使う場合は、画面上で入力するとか

**Reply by kagomen** [reply]
> テーマによって、好きな色が白とか黒だと、背景や文字の色に使われて、周りの色と同化して見えなくなるとか

ここらへんは（ユーザー側で自由に調整するよりは不自由なんですが）、テーマ側で調整できると良さそうかな？と思いました
例えば選択した色が明度n以上だったら文字の影色をx色にする、とか…？
あとは、その少しの不便さによって、ユーザーが自作テーマ開発するモチベーションにつながったらいいなー……という気持ちもあります

> Contributions.jsonには、好きな絵文字を１つ入力して、複数の絵文字を使う場合は、画面上で入力するとか

技術的に難しい場合は、改めて考えた方が良さそうですね！

**Reply by kagomen** [reply]
ひとまず、ちょっと頭の中にあるデザイン案を起こしてみます！
